### ISSUE TITLE
Cannot select an element inside V8 combobox in a V14 dialog

### ISSUE THREAD

**Issue by jcgueriaud1** [vaadin/multiplatform-runtime#63]
Hi,

I cannot select a value inside V8 combobox with the mouse if the combobox is in a V14 Dialog.

Steps to reproduce:
* Open a V14 Dialog with a V8 combobox.
* Click on the combobox
* Value can be selected with the keyboard but not with the mouse

This is probably because the V8 overlay is inside the body and not inside the vaadin-dialog-overlay.

One workaround is to add this in your main style:

.v-overlay-container {
    pointer-events: auto;
}

**Comment by jcgueriaud1** [comment]
I don't know if it's a bug, but it may be documented as a limitation (if the workaround does not work as expected) and also V8 Window/ V14 Dialog "incompatibility".

**Comment by jcgueriaud1** [comment]
You will have the same problem for a Vaadin 7/8 Datefield.

The dialog will close when you choose a date if the dialog is setCloseOnOutsideClick(true).

It's because the V7/8 overlay is outside the dialog (from a dom perspective).
As a workaround you can setCloseOnOutsideClick(false)

**Comment by TatuLund** [comment]
This is a documented limitation: https://vaadin.com/docs/latest/tools/mpr/configuration/limitations#z-indexes

**Comment by jcgueriaud1** [comment]
The main issue is that the Vaadin 8 overlay is outside the vaadin-dialog overlay then it's blocked when the dialog is modal.

The z-index is not really a problem (since you can clearly see the Vaadin 8 overlay).

Technically it's not a limitation of MPR but a limitation of the dialog that doesn't allow any other overlays (from a third party or Vaadin 8)

**Comment by TatuLund** [comment]
> Technically it's not a limitation of MPR but a limitation of the dialog that doesn't allow any other overlays (from a third party or Vaadin 8)

Yes, that is correct.

> blocked when the dialog is modal.

Blocked by client side curtain or server side block?

**Comment by TatuLund** [comment]
> Blocked by client side curtain or server side block?

Well, in Vaadin 14 there is no server side modality ... But with Vaadin 24 there is.

**Comment by TatuLund** [comment]
> The z-index is not really a problem (since you can clearly see the Vaadin 8 overlay).

The documentation could be a bit better as it does not explicitly mention the modal curtain problem.

**Comment by TatuLund** [comment]
Note, with Vaadin 24 one needs to use this workaround as well:

https://github.com/vaadin/multiplatform-runtime/issues/122#issuecomment-1864445151

**Comment by TatuLund** [comment]
Since Vaadin 8.27.4 add this CSS is no longer needed as it has been included in Vaadin 8:

```
.v-overlay-container {
  pointer-events: auto;
}
```
